### PR TITLE
chore(flightsql): Add tests that schema and data matches

### PR DESCRIPTION
**Pull request by alamb**
Part of https://github.com/influxdata/influxdb_iox/issues/7278

This introduces a test that the schema and results for FlightSQL endpoints are the same. In my next PR for https://github.com/influxdata/influxdb_iox/issues/7278 I plan to hard code the schemas and want to ensure that they are consistent with the data

It turns out to have been a good exercise, as I also found another bug: https://github.com/influxdata/influxdb_iox/issues/7278

### CODE REVIEW

File: influxdb_iox/tests/end_to_end_cases/flightsql.rs [hunk 1]
```diff
@@ -625,6 +633,100 @@ async fn flightsql_jdbc() {
...
+    while let Some(batch) = result_stream.try_next().await.unwrap() {
+        saw_data = true;
+        assert_eq!(batch.schema().as_ref(), &flight_info_schema);
+        // The stream itself also may report a schema
+        if let Some(stream_schema) = result_stream.schema() {
+            assert_eq!(stream_schema.as_ref(), &flight_info_schema);
+        }
+    }
+    // should have seen at least one point
+    assert!(saw_data);
```
Comment (appletreeisyellow) [inline influxdb_iox/tests/end_to_end_cases/flightsql.rs:727]: Does `saw_data` check for there is at lease one result from `DoGet` call?

**Review by appletreeisyellow — APPROVED** [review]
It's awesome that you caught a bug with the added test! I left one question below

File: influxdb_iox/tests/end_to_end_cases/flightsql.rs [hunk 2]
```diff
@@ -625,6 +633,100 @@ async fn flightsql_jdbc() {
...
+    let mut saw_data = false;
+    while let Some(batch) = result_stream.try_next().await.unwrap() {
+        saw_data = true;
+        assert_eq!(batch.schema().as_ref(), &flight_info_schema);
+        // The stream itself also may report a schema
+        if let Some(stream_schema) = result_stream.schema() {
+            assert_eq!(stream_schema.as_ref(), &flight_info_schema);
+        }
+    }
+    // should have seen at least one point
```
Comment (alamb) [inline influxdb_iox/tests/end_to_end_cases/flightsql.rs:726]:
```suggestion
    // verify we have seen at least one RecordBatch
    // (all FlightSQL endpoints return at least one)
```

**Comment by alamb** [comment]
I rebased to fix the semantic CI check (the commit message did not pass)

**Pull request merged**